azure/datalake/store/lib.py

```diff
@@ -141,23 +143,43 @@ def auth(tenant_id=None, username=None,
                                                             client_secret)
     else:
         raise ValueError("No authentication method found for credentials")
+
     out.update({'access': out['accessToken'], 'resource': resource,
                 'refresh': out.get('refreshToken', False),
                 'time': time.time(), 'tenant': tenant_id, 'client': client_id})
-    return out
 
+    return DataLakeCredential(out)
+
+class DataLakeCredential(Authentication):
```

Comment:
@lmazuel we briefly synced separately on this. I will work with you a bit later to see if we can further generalize the Auth so that I can enable the ability to use any signed_session() enabled credential.